### PR TITLE
feat(assets): add getAssets and isAssetUnlinked methods

### PR DESCRIPTION
**Pull request by gellis**

### CODE REVIEW

**Comment by shieldo** [comment]
@gellis Think there's some PHP7 only syntax in here - this library is still supporting PHP 5.6 at the current time.

File: src/Contentful.php [hunk 1]
```diff
@@ -274,6 +276,59 @@ public function getAssetAsync($id, $space = null, array $options = [], $locale =
...
+            $filters
+        );
+    }
+
+    /**
+     * @param string|SpaceInterface $space
+     * @param string $assetId
+     * @return bool
+     */
+    public function isAssetUnlinked($space, string $assetId)
```
Comment (shieldo) [inline src/Contentful.php]: Library is still supporting PHP 5.6 at the current time, so we cannot use scalar type hints here just yet.

File: src/Contentful.php [hunk 2]
```diff
@@ -274,6 +276,59 @@ public function getAssetAsync($id, $space = null, array $options = [], $locale =
...
+    {
+        $spaceName = ($space instanceof SpaceInterface) ? $space->getName() : $space;
+        $spaceData = $this->getSpaceDataForName($spaceName);
+        $api = ($spaceData['preview_mode']) ? self::PREVIEW_API : self::CONTENT_DELIVERY_API;
+        $filters = [];
+        $filters[] = new LinksToAssetFilter($assetId);
+
+       $linkedEntries = $this->doRequest(
```
Comment (shieldo) [inline src/Contentful.php]: Indentation here is slightly out.

File: src/Contentful.php [hunk 3]
```diff
@@ -274,6 +276,59 @@ public function getAssetAsync($id, $space = null, array $options = [], $locale =
...
+    }
+
+    /**
+     * @param string|SpaceInterface $space
+     * @param string $assetId
+     * @return bool
+     */
+    public function isAssetUnlinked($space, string $assetId)
+    {
+        $spaceName = ($space instanceof SpaceInterface) ? $space->getName() : $space;
```
Comment (shieldo) [inline src/Contentful.php]: This method could be a lot simpler and easier to understand if it called down onto `getEntries` instead.

File: src/Contentful.php [hunk 4]
```diff
@@ -274,6 +276,59 @@ public function getAssetAsync($id, $space = null, array $options = [], $locale =
         );
     }
 
+    /**
+     * @param string|SpaceInterface $space
+     * @return PromiseInterface|AssetInterface
+     */
+    public function getAssets($space)
```
Comment (shieldo) [inline src/Contentful.php]: A method to get multiple assets here should follow the same signature as `getEntries`, and there should also be a `getAssetsAsync` method similarly. The implementation here fetches all assets (up to a limit of 1000) which seems to have a particular purpose in mind - this library is pretty generic and unopinionated about uses, so really any filtering should be passed in from the outside.

File: src/Contentful.php [hunk 5]
```diff
@@ -274,6 +275,61 @@ public function getAssetAsync($id, $space = null, array $options = [], $locale =
...
+    public function getAssets(array $parameters = [], $space = null, array $options = [])
+    {
+        $spaceName = ($space instanceof SpaceInterface) ? $space->getName() : $space;
+        $spaceData = $this->getSpaceDataForName($spaceName);
+        $api = ($spaceData['preview_mode']) ? self::PREVIEW_API : self::CONTENT_DELIVERY_API;
+
+        return $this->doRequest(
+            $spaceData,
+            $spaceName,
+            $this->getEndpointUrl(sprintf('/spaces/%s/assets/', $spaceData['key']), $api),
```
Comment (shieldo) [inline src/Contentful.php]: One minor thing - the API doesn't specify a trailing slash on the URL here (there isn't one for the entries in `getEntries()` either) - can this be removed?

**Comment by gellis** [comment]
@shieldo Sure thing, slash is now removed 👍

**Comment by shieldo** [comment]
@gellis Thanks for this contribution!

**Pull request merged**